CHANGELOG.md

```diff
@@ -30,6 +30,7 @@
 
 ### 💡 Enhancements 💡
 
+- `mdatagen`: Adds support for declaring feature gates in metadata.yaml file. (#21801)
```

Comment:
Changelog needs to be added to `./.chloggen`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry